### ISSUE TITLE
Any plans to support a water flow sensor via the Octo?

### ISSUE THREAD

**Issue by mcronce** [aleksamagicka/aquacomputer_d5next-hwmon#6]

**Comment by aleksamagicka** [comment]
Yes, but I don't have such a sensor to connect to the Octo, so I don't know the register offset nor if it requries calibration or such. If you or someone else has it, and is willing to tinker a bit and capture HID reports, just say so and I'll write a small tutorial for that.

**Comment by mcronce** [comment]
I have one that I haven't installed yet because there's no driver support :laughing: 

I'm close to due for loop maintenance, though; I can add it while I have everything drained and report back.  I probably won't get to it next weekend, but the weekend after is likely.

**Comment by aleksamagicka** [comment]
So the way to get the register offsets is to capture the HID reports while simultaneously writing down/noting the values from Aquasuite or Hwinfo64 (this requires Windows). All Aquacomuter devices send a sensor report each second. Once you have a sample HID report, open it in a hex editor and search for the flow value as it's shown in Aquasuite, in big endian. Chances are that you won't find the exact value (for example, Aquasuite shows 27.4C, and the report actually says 2745), so you'll need to explore a bit.

We're not on a timeline, do it whenever you can 😄

**Comment by JackDoan** [comment]
there's probably also some setup messages we need to send to tell the Octo "hey, flow sensor is here"
